### PR TITLE
Automatedly advance ObjC version numbers

### PR DESCRIPTION
**Pull request by muxi**
Generate ObjC podspec version numbers using template

Advance BoringSSL version to 8.0 in master

### CODE REVIEW

File: src/objective-c/GRPCClient/private/version.h [hunk 1]
```diff
@@ -31,4 +31,7 @@
  *
  */
 
+/* This file is autogenerated from template:
+ * templates/src/objective-c/GRPCClient/private/version.h.template */
```
Comment (jcanizales) [inline src/objective-c/GRPCClient/private/version.h]: How can it be regenerated?

**Comment by muxi** [comment]
And yes it does it automatically. No need to add the files to somewhere. `generate_projects.sh` checks all files by traversing the `templates/` dir.

File: src/objective-c/GRPCClient/private/version.h [hunk 2]
```diff
@@ -31,7 +31,10 @@
  *
  */
 
-/* This file is autogenerated from template:
- * templates/src/objective-c/GRPCClient/private/version.h.template */
+/* This file is autogenerated from a template file. Please make
+ * modifications to
+ * `templates/src/objective-c/GRPCClient/private/version.h.template`
+ * instead. This file can be regenerated from the template by running
+ * `tools/buildgen/generate_projects.sh`. */
```
Comment (jcanizales) [inline src/objective-c/GRPCClient/private/version.h]: Another nit: I *think* I remember the style being followed in the ObjC files for block comments is:
```objc
/*
 * blahblah
 */
```

Comment (muxi) [inline src/objective-c/GRPCClient/private/version.h]: I found most of ObjC files followed the style of double slashes for comments other than the license. I'll follow that.

**Comment by muxi** [comment]
Please approve #9480 as well :)

**Comment by muxi** [comment]
Test failures:
Basic Tests Linux - #9437 
gRPC_performance_pull_requests - #9499

**Pull request merged**